_drgn.pyi

```diff
@@ -733,6 +733,50 @@ class Program:
         :param lang: :attr:`Type.language`
         """
         ...
+    def threads(self) -> Iterable[Thread]:
+        """
+        Return an iterable over all of the threads resident within the process.
+        """
+        ...
+    def thread(self, tid: IntegerLike) -> Thread:
+        """
+        Retrieve the thread with the given thread ID.
+
+        :param tid: The desired thread's ID (as defined by `gettid(2) <http://man7.org/linux/man-pages/man2/gettid.2.html>`_)
+        :raises LookupError: if no such thread exists
+        """
+        ...
+    def crashed_thread(self) -> Thread:
+        """
+        Return the thread responsible for causing the process to crash (e.g. the thread that segfaulted,
+        received a `SIGQUIT`, etc.). This method should only be called when debugging a core dump.
+
+        :raises ValueError: if the current target is not a core dump
+        """
+        ...
+
+class Thread:
+    """
+    A single thread resident within a process.
+    """
+
+    tid: int
+    """ This thread's ID (as defined by `gettid(2) <http://man7.org/linux/man-pages/man2/gettid.2.html>`_). """
+    prstatus: bytes
+    """ The contents of the PRSTATUS note associated with this thread in the core dump. """
```

Comment:
I'd rather add getters and/or attributes for interesting information that can be found in `prstatus`, so let's remove this.